tests/components/pglab/test_config_flow.py

```diff
@@ -0,0 +1,94 @@
+"""Test the PG LAB Electronics config flow."""
+
+from homeassistant.components.pglab.const import DOMAIN
+from homeassistant.config_entries import SOURCE_MQTT
+from homeassistant.core import HomeAssistant
+from homeassistant.data_entry_flow import FlowResultType
+from homeassistant.helpers.service_info.mqtt import MqttServiceInfo
+
+from tests.common import MockConfigEntry
+from tests.typing import MqttMockHAClient
+
+
+async def check_single_instance_configuration(
+    hass: HomeAssistant, mqtt_mock: MqttMockHAClient, source: str
+):
+    """Check if PGLab configuration is allowed for multiple instances."""
+    MockConfigEntry(domain=DOMAIN).add_to_hass(hass)
+
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": source}
+    )
+
+    # Be sure that result is abort. Only single instance is allowed.
+    assert result["type"] is FlowResultType.ABORT
+    assert result["reason"] == "single_instance_allowed"
+
+
+async def test_mqtt_config_single_instance(
+    hass: HomeAssistant, mqtt_mock: MqttMockHAClient
+) -> None:
+    """Test MQTT flow aborts when an entry already exist."""
+    await check_single_instance_configuration(hass, mqtt_mock, SOURCE_MQTT)
```

Comment:
Why is this split?

Reply:
Not anymore needed...  It was relevant in a previous patch.
I'm going to avoid the split.